barretenberg/cpp/src/barretenberg/vm/avm_trace/AvmMini_deserialization.cpp

```diff
@@ -0,0 +1,176 @@
+            }
+            uint8_t set_tag_u8 = bytecode.at(pos);
+            if (!std::set<uint8_t>{ static_cast<uint8_t>(AvmMemoryTag::U8),
+                                    static_cast<uint8_t>(AvmMemoryTag::U16),
+                                    static_cast<uint8_t>(AvmMemoryTag::U32),
+                                    static_cast<uint8_t>(AvmMemoryTag::U64),
+                                    static_cast<uint8_t>(AvmMemoryTag::U128) }
+                     .contains(set_tag_u8)) {
+                throw_or_abort("Instruction tag for SET opcode is invalid at position " + std::to_string(pos) +
+                               " value: " + std::to_string(set_tag_u8));
+            }
+
+            auto in_tag = static_cast<AvmMemoryTag>(set_tag_u8);
+            switch (in_tag) {
+            case AvmMemoryTag::U8:
+                inst_format = { OperandType::TAG, OperandType::UINT8, OperandType::UINT32 };
+                break;
+            case AvmMemoryTag::U16:
+                inst_format = { OperandType::TAG, OperandType::UINT16, OperandType::UINT32 };
+                break;
+            case AvmMemoryTag::U32:
+                inst_format = { OperandType::TAG, OperandType::UINT32, OperandType::UINT32 };
+                break;
+            case AvmMemoryTag::U64:
+                inst_format = { OperandType::TAG, OperandType::UINT64, OperandType::UINT32 };
+                break;
+            case AvmMemoryTag::U128:
+                inst_format = { OperandType::TAG, OperandType::UINT128, OperandType::UINT32 };
+                break;
+            default: // This branch is guarded above.
+                inst_format = {};
+            }
+        } else {
+            inst_format = OPCODE_WIRE_FORMAT.at(opcode);
+        }
+
+        std::vector<Operand> operands{};
+
+        for (auto const& opType : inst_format) {
+            // No underflow as while condition guarantees pos <= length (after pos++)
+            if (length - pos < OPERAND_TYPE_SIZE.at(opType)) {
+                throw_or_abort("Operand is missing at position " + std::to_string(pos));
+            }
+            Operand operand;
+            uint8_t const* pos_ptr = &bytecode.at(pos);
+
+            switch (opType) {
+            case OperandType::TAG: {
+                uint8_t tag_u8 = bytecode.at(pos);
+                if (tag_u8 == static_cast<uint8_t>(AvmMemoryTag::U0) || tag_u8 > MAX_MEM_TAG) {
+                    throw_or_abort("Instruction tag is invalid at position " + std::to_string(pos) +
+                                   " value: " + std::to_string(tag_u8));
+                }
+                operand = static_cast<AvmMemoryTag>(tag_u8);
+                pos++;
+                break;
+            }
+            case OperandType::UINT8:
+                operand = bytecode.at(pos);
+                pos++;
+                break;
+            case OperandType::UINT16: {
+                uint16_t operand_u16{};
```

Comment:
`{}` not only not needed but highly discouraged :)

Reply:
I get some warning that operand_u16 is unitialized if I remove the {}. Same for the other instances of the same flavour.

Reply:
Yes, that's right. The thing is that `{}` is really strange for most types but specially for primitive types. If you need to initialize (because VS or the compiler is dumb like now) it's better to explicitly do `= 0`.